src/components/TodoAddForm/TodoAddForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from 'react';
+import { PreparedTodo } from '../../types/PreparedTodo';
+import { User } from '../../types/User';
+import './TodoAddForm.scss';
+
+interface Props {
+  todos: PreparedTodo[],
+  users: User[],
+  onAdd: (todo: PreparedTodo) => void,
+}
+
+export const TodoAddForm: React.FC<Props> = ({ todos, users, onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [isTitleValid, setIsTitleValid] = useState(true);
+  const [isUserIdValid, setIsUserIdValid] = useState(true);
+
+  const validatorABC = /[^a-zA-Zа-яА-Я0-9\s]/g;
+
+  const handleTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(validatorABC, ''));
+    setIsTitleValid(true);
+  };
+
+  const handleUserIdInput = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setIsUserIdValid(true);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!(title.length)) {
+      setIsTitleValid(false);
+    }
+
+    if (!userId) {
+      setIsUserIdValid(false);
+    }
+
+    if (title.length && userId) {
```

Comment:
Better to do `if (!title.length && !userId) { return ;}`

Reply:
Not fixed. Better not to write main logic in `if` block. If you have questions about that, feel free to ask